src/trim_primer_quality.cpp

```diff
@@ -516,9 +530,16 @@ int trim_bam_qual_primer(std::string bam, std::string bed, std::string bam_out,
     std::cout << cit->get_name() << "\t" << cit->get_read_count() << std::endl;
   }
   std::cout << std::endl << "Trimmed primers from " << round_int(primer_trim_count, mapped) << "% (" << primer_trim_count <<  ") of reads." << std::endl;
-  std::cout << round_int( low_quality, mapped) << "% (" << low_quality << ") of reads were quality trimmed below the minimum length of " << min_length << " bp and were not writen to file." << std::endl;
+  std::cout << round_int( low_quality, mapped) << "% (" << low_quality << ") of reads were quality trimmed below the minimum length of " << min_length << " bp and were ";
+  if (mark_qcfail_flag) {
+    std::cout << "marked as failed" << std::endl;
+  } else {
+    std::cout << "not written to file." << std::endl;
+  }
   if(write_no_primer_reads){
-    std::cout << round_int(no_primer_counter, mapped) << "% ("  << no_primer_counter << ") of reads started outside of primer regions. Since the -e flag was given, these reads were written to file." << std::endl;
+    std::cout << round_int(no_primer_counter, mapped) << "% ("  << no_primer_counter << ") of reads started outside of primer regions. Since the -e flag was given, these reads were written to file";
+    if (mark_qcfail_flag) std::cout << " and the BAM_QCFAIL flag set";
```

Comment:
The intent of this "mark as QCFAIL" option is to produce a BAM useful for _both_ further processing by say `ivar consensus` or `ivar variants` (via `samtools mpileup` which I believe will drop QCFAIL records by default), _and_ creating fastq (with a command that does not strip QCFAIL) for reprocessing (by an updated or different pipeline).

Also, I'd argue that with a library type where we expect to see primers (e.g. Illumina adapter ligated to the amplicon) that the primer's absence _could_ be interpreted as a type of QC fail.

Would updating variable names e.g. `write_no_primer_reads`, and adding/altering explanatory text help?

Reply:
Sure, I agree that in some cases this will be a type of QC fail. However, my concern here is if there is a NextEra step and the user explicitly asks to write reads with no primer to file, we would want them to be used for consensus generation and variant calling. I would want to differentiate these from say reads that didn't pass the minimum length threshold. 

How about if `-e` if specified, then the reads are not marked as QC fail else they are?

Reply:
Okay sure, I see your point. In any case, using `-e` without `-f` will retain the current behavior. I will document this as well.